### PR TITLE
Mechs must now have maint protocols active to be repaired.

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

This PR adds a check to welder_act on mech code, requiring maintenance protocols to be active in order for a mech to be repaired via welder.

## Why It's Good For The Game

Mechs are strong, perhaps too strong, and could use some nerfs. This is a simple change, that forces a mech to be vulnerable, and unable to move or fire, when getting welder repaired. This prevents crew from fighting blob with a mech and constantly welding it, making it effectively an un-killable infinitely firing turret. Instead, the mech must retreat a distance to be fixed and stop firing, or be repaired on the front lines and vulnerable.

## Changelog
:cl:
tweak: Mechs must now have maintenance protocols active in order to be repaired with a welder.
/:cl:

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -840,6 +840,9 @@
 	if(repairing)
 		to_chat(user, "<span class='notice'>[src] is currently being repaired!</span>")
 		return
+	if(state < 1)
```
Comment (moxian) [inline code/game/mecha/mecha.dm:843]:
```suggestion
	if(state == 0)
```
The state numbers are just arbitrary unordered magic numbers. It does not make sense to compare them for greater than/less than, just as it doesn't make sense to compare `"normal mode"` `"maintenance mode"` `"power cell removed"` etc. via `<`.

**Comment by procdrone** [comment]
cannot fight while repairing, sounds good. if damaged, the pilot will be forced to either retreat and fix or take a gamble and continue fighting despite damage and risk loosing the mech.

**Comment by TDSSS** [comment]
It might be worth thinking about buffing nukie mechs, or making them cheaper, along with this change. They're already not very viable or used very often, and this does make them even less attractive of a purchase.

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -840,6 +840,9 @@
 	if(repairing)
 		to_chat(user, "<span class='notice'>[src] is currently being repaired!</span>")
 		return
+	if(state == 0)
```
Comment (SteelSlayer) [inline code/game/mecha/mecha.dm:843]: I think you want to use `maint_access` here for clarity. If its `1`, it means maint protocols are on, `0` is off. Mech code is so convoluted, it's hard to tell what state 0 - 4 are suppose to do at a glance.

**Review by SteelSlayer — APPROVED** [review]

**Review by Kyep — APPROVED** [review]

File: code/game/mecha/mecha.dm (hunk 2 above)
Comment (farie82) [inline code/game/mecha/mecha.dm:843]: Ah yes state... It'll be out of scope to refactor this into defines but can you place a comment saying what the state means? Makes it easier for later

File: code/game/mecha/mecha.dm [hunk 3]
```diff
@@ -840,6 +840,9 @@
...
+		to_chat(user, "<span class='notice'>[src] can not be repaired without maintenance protocols active!</span>")
```
Comment (farie82) [inline code/game/mecha/mecha.dm:844]:
```suggestion
		to_chat(user, "<span class='warning'>[src] can not be repaired without maintenance protocols active!</span>")
```
I'd make this a warning level message

**Review by farie82 — CHANGES_REQUESTED** [review]

File: code/game/mecha/mecha.dm (hunk 2 above)
Comment (Qwertytoforty) [reply code/game/mecha/mecha.dm]: Done!

**Review by farie82 — APPROVED** [review]
LGTM

**Pull request merged**